typing/typecore.ml

```diff
@@ -3041,7 +3041,11 @@ and type_pat_aux
   | Ppat_array (mut, spl) ->
       let mut =
         match mut with
-        | Mutable -> Mutable Alloc.Comonadic.Const.legacy
+        | Mutable  -> Mutable {
+          modal_upper_bound = Alloc.Comonadic.Const.legacy;
+          (* CR aspsmith: Revisit once we support atomic arrays *)
```

Comment:
What is the status of atomic array elements, anyway? Is there upstream support? Is it planned? I'm wondering if there should be an internal epic for this... and if there is, then we can label this comment with the epic somehow so we can find all such comments.

Reply:
I think we can basically support it with just some stdlib changes after this series of PRs? And I plan on doing that, since I have some use cases for them.